java/flight/flight-core/src/main/java/org/apache/arrow/flight/FlightServer.java

```diff
@@ -328,6 +360,15 @@ public Builder useTls(final InputStream certChain, final InputStream key) {
       return this;
     }
 
+    /**
+     * Enable mTLS on the server.
+     * @param mTlsCACert The CA certificate to use for verifying clients.
+     */
+    public Builder useMTlsClientVerification(final InputStream mTlsCACert) throws IOException {
+      this.mTlsCACert = mTlsCACert;
```

Comment:
Similar comment as above about this potentially leaking a file. Also, now it is ambiguous whether the Flight client should close() mTlsCACert (presumably it should).

Reply:
I mirrored the existing `useTls` method [here](https://github.com/apache/arrow/blob/fcbcb7dab71b14ff3436af7b21bd23132877e2fa/java/flight/flight-core/src/main/java/org/apache/arrow/flight/FlightServer.java#L315)

Does that code have the issue as well?

Reply:
Correct. It looks to have this issue too.

Reply:
@vibhatha @davisusanibar can you file an issue to fix this?

Reply:
@lidavidm I will file one. Will study a bit and file one.

Reply:
given that this issue will be filed separately - is it acceptable to approve/merge this PR?

Reply:
Filed #38586